lang/pypy/Portfile

```
@@ -1,26 +1,43 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           gitlab 1.0
+PortGroup           github 1.0
 PortGroup           select 1.0
-PortGroup           deprecated 1.0
 PortGroup           openssl 1.0
 
-gitlab.instance     https://foss.heptapod.net
-gitlab.setup        pypy pypy 7.3.9
+# Define PyPy versions for each Python branch - update these pairs together
+array set pypy_versions {
+    2.7   7.3.20
+    3.8   7.3.11
+    3.9   7.3.16
+    3.10  7.3.19
+    3.11  7.3.20
+}
+
+# Set default to latest offered Python version: 3.11
+set python.branch   3.11
+version             $pypy_versions(${python.branch})
+
+# Use github.setup to configure the portgroup for livecheck,
+# then override the variables it sets for fetching.
+github.setup        pypy pypy ${version} release-pypy${python.branch}-v
+
+# Override distname and master_sites to match the actual release asset URL
+distname            ${github.tag_prefix}${version}
+master_sites        https://github.com/pypy/pypy/archive/refs/tags/
+extract.suffix      .tar.gz
```

You should not need to override these; you should just need to set `github.tarball_from`.

```suggestion
github.tarball_from archive
```